packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -165,6 +192,34 @@ export class PendingStateManager {
         this.pendingStates.push(pendingFlush);
     }
 
+    public processRemoteMessage(message: ISequencedDocumentMessage) {
+        while (!this.initialStates.isEmpty()) {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const nextMessage = this.initialStates.peekFront()!;
+            assert(nextMessage.type === "message"); // assume it's not flush for now
+            if (nextMessage.referenceSequenceNumber > message.sequenceNumber) {
```

Comment:
Please add assert after that if that nextMessage.referenceSequenceNumber === message.sequenceNumber, that will catch the case if we loaded from later snapshot and had no chance to observe proper ops